saltstack/base/salt/prerequisites/usr/bin/user-data-helper.sh

```diff
@@ -253,6 +259,11 @@ resize_partitions() {
 }
 
 main() {
+  if [[ "$SECRET_ENCRYPTION_ENABLED" == "true" ]]; then
+    echo "$SECRET_ENCRYPTION_KEY_SOURCE" 1> /etc/luks/passphrase_encryption_key
+    create_luks_volume
+    # then move secrets to LUKS volume and symlink them to their original place
```

Comment:
And also extract the EC2 user data secrets and store them either as ciphertext on the root disk, or as plaintext inside the LUKS volume. 😄